test/net/sourceforge/kolmafia/request/RequestTestBase.java

```diff
@@ -1,22 +1,72 @@
 package net.sourceforge.kolmafia.request;
 
+import static org.mockito.ArgumentMatchers.argThat;
 import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+import net.sourceforge.kolmafia.extensions.ForbidNetworkAccess;
+import net.sourceforge.kolmafia.utilities.HttpUtilities;
+import net.sourceforge.kolmafia.utilities.HttpUtilities.ConnectionFactory;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.mockito.ArgumentMatcher;
+import org.mockito.verification.VerificationMode;
 
+/**
+ * Extend this class when making tests involving {@link GenericRequest} and/or its descendants.
+ *
+ * <ul>
+ *   <li>Allows this class's tests to run {@link GenericRequest}s (otherwise blocked pre-login and
+ *       due to {@link ForbidNetworkAccess}).
+ *   <li>For {@link GenericRequest}s made locally, gives access to {@link
+ *       RequestTestBase#expectSuccess}, allowing the simulation of running a {@link
+ *       org.mockito.Mockito#spy spy} of that request.
+ *   <li>For the rest, gives access to {@link #respondIfContains}/{@link #respondIfMatches}, causing
+ *       future executed requests to succeed with the given response if the path of the URL
+ *       submitted contains the submitted text(s)/matches the submitted pattern(s)/matches the
+ *       submitted {@link ArgumentMatcher}. These methods all return the {@link ArgumentMatcher}
+ *       submitted for the stubbing. This allows them to be re-used to {@link
+ *       org.mockito.Mockito#verify verify} the access of the method (using the {@link
+ *       #verifyRequest} shortcut method).
```

Comment:
You shouldn't normally need to directly verify() your requests. Instead, prefer to use strict stubs (which are planned to be the default in Mockito 4).